File: API/Entities/RegistrationRequest.cs
namespace API.Entities;

public class RegistrationRequest
{
    public int Id { get; set; }

    public string OwnerFirstName { get; set; } = null!;
    public string OwnerLastName { get; set; } = null!;
    public string OwnerPhone { get; set; } = null!;
    public string OwnerEmail { get; set; } = null!;

    public string ObjectName { get; set; } = null!;
    public int ObjectTypeId { get; set; }
    public int MunicipalityId { get; set; }
    public ObjectType ObjectType { get; set; } = null!;
    public Municipality Municipality { get; set; } = null!;
    public string Address { get; set; } = null!;

    public string Status { get; set; } = null!;

    public DateTime CreatedAt { get; set; }
}

File: client/src/models/types/RegistrationRequest.ts
export interface RegistrationRequestDto {
  ownerFirstName: string;
  ownerLastName: string;
  ownerPhone: string;
  ownerEmail: string;
  objectName: string;
  objectTypeId: number;
  municipalityId: number;
  address: string;
}

export interface GetRegistrationRequestDto {
  id: number;
  ownerFirstName: string;
  ownerLastName: string;
  ownerPhone: string;
  ownerEmail: string;
  objectName: string;
  objectType: string;
  municipality: string;
  address: string;
  status: string;
  createdAt: string; 
}

export interface UpdateStatusDto {
  status: string;
}